arduino-ide-extension/src/node/monitor-manager.ts

```diff
@@ -56,13 +95,35 @@ export class MonitorManager extends CoreClientAware {
    * @returns a Status object to know if the process has been
    * started or if there have been errors.
    */
-  async startMonitor(board: Board, port: Port): Promise<Status> {
+  async startMonitor(
+    board: Board,
+    port: Port,
+    postStartCallback: (status: Status) => void
```

Comment:
this callback is passed in rather than executed on ".then" after invocation, this is because if an upload is in progress we add the callback to our "queue" "this.startMonitor_PendingRequests" and return